lib/get-category-path.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var path = require('path'),
+  utility = require('./utility');
+
+
+/**
+ * Determines the category directory structure from the pattern's data
+ *
+ * @param {Object} patternObject  individual pattern's data (from /pattern-name/pattern.yml)
+ * @param {Object} options  pattern-importer options
+ *
+ * @return {String} categoryPath  internal path url to category
+ *
+ * @requires ./lib/utility
+ * @requires ./lib/category-name-converter
+ */
+exports.getCategoryPath = function (patternObject, options) {
```

Comment:
@e2tha-e A major overhaul? Can you be more specific? That seems very drastic. I take that to mean I'm making some major mistakes in proper coding standards. Is that the case?

@ericduran deep nested if statements have been removed in this file. please re-check

For the second point - there is a lot of code, yes. But the majority of this has been split out because 
1. I wanted to unit test individual functionality
2. Anytime I felt a piece of functionality could get used more than once, I broke it out to keep the system dry.
3. I wanted to allow us to use these pieces of code outside of this system